dot_config/nvim/lua/tap/plugins/lualine.lua

```diff
@@ -280,40 +245,28 @@ return {
         lualine_z = { { '%p%%', cond = conditions.is_wide_window } },
       }
 
-      require('tap.utils').apply_user_highlights(
-        'Lualine',
-        function(highlight, _, lsp_color)
-          highlight('LualineDiagnosticError', {
-            guibg = lsp_color 'error',
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-          highlight('LualineDiagnosticWarn', {
-            guibg = lsp_color 'warning',
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-          highlight('LualineDiagnosticHint', {
-            guibg = lsp_color 'hint',
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-          highlight('LualineDiagnosticInfo', {
-            guibg = lsp_color 'info',
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-          highlight('LualineDiagnosticOk', {
-            guibg = lsp_color 'ok',
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-          highlight('NavicSeparator', {
-            guifg = color { dark = 'nord3_gui', light = 'fg' },
-          })
-
-          local theme_name = vim.g.colors_name
-          local theme = theme_name == 'nord' and nord_theme or theme_name
-          if theme then
-            require('lualine').setup { options = { theme = theme } }
-          end
-        end
-      )
+      require('tap.utils').apply_user_highlights('Lualine', function(highlight)
+        highlight('LualineDiagnosticError', {
+          guibg = highlight_group_attrs('DiagnosticError').guifg,
+          guifg = highlight_group_attrs('lualine_c_normal').guibg,
```

Comment:
This would probably be nicer if I just used the palette provided in the callback function to `apply_user_highlights`